test/functional/write_concern.test.js

```diff
@@ -117,4 +117,56 @@ describe('Write Concern', function () {
         return client.close();
       });
   });
+
+  describe('details property', () => {
+    let client;
+
+    beforeEach(async function () {
+      client = this.configuration.newClient({ monitorCommands: true });
+      await client.connect();
+    });
+
+    afterEach(async () => {
+      if (client) await client.close();
+    });
+
+    it('should provide details on why document validation failed', {
+      metadata: { requires: { mongodb: '>=5.0.0' } },
+      async test() {
+        try {
+          await client.db().collection('wc_details').drop();
+        } catch {
+          // don't care
+        }
+
+        const collection = await client
+          .db()
+          .createCollection('wc_details', { validator: { x: { $type: 'string' } } });
+
+        const evCapture = require('events').once(client, 'commandSucceeded');
+
+        let insertError;
+        try {
+          await collection.insertOne({ x: /not a string/ });
+          expect.fail('The insert should fail the validation that x must be a string');
+        } catch (error) {
+          expect(error).to.be.instanceOf(MongoServerError);
+          expect(error).to.have.property('code', 121);
+          expect(error).to.have.nested.property('errInfo.details').that.is.an('object');
```

Comment:
the way I am reading the spec, it says that the `errInfo` property is what should be exposed as `error.details` in the driver; what we currently do is expose `errInfo` as `error.errInfo`, it also says we shouldn't try to parse the inside of `errInfo` so accessing `errInfo.details` probably should not be in the test

Reply:
Did a bit of hunting on this one: 

- [Looks like we need to leave errInfo as is](https://jira.mongodb.org/browse/DRIVERS-820?focusedCommentId=3604057&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-3604057)
- [The spec permits us to have errInfo on generic Server write failures](https://github.com/mongodb/specifications/blob/master/source/server_write_commands.rst#error-document)
- [Following PHPs lead here](https://github.com/mongodb/mongo-php-driver/pull/1229/files#diff-f5695d70f8e6c3e6adedcd927165379ee9dbf1be9284dd219473db5b077a2eb8R43) I agree with removing the parsing of errInfo I'll push that up in the next commit

Reply:
Thanks for the thorough check!